test/detection/test_core.py

```diff
@@ -1,5 +1,5 @@
 from contextlib import ExitStack as DoesNotRaise
-from typing import List, Optional, Union
+from typing import Optional, Union
```

Comment:
lets replace `Union` with `|` and `Optional` with `| None` to keep things consistent